### PR TITLE
Feat/oauth meinbildungsraum

### PR DESCRIPTION
**Pull request by timber-they**
Add OIDC backend. Also note the ui PR!

### CODE REVIEW

**Comment by timber-they** [comment]
UI PR: https://github.com/mint-o-badges/badgr-ui/pull/406

**Review by jona159 — APPROVED** [review]

File: apps/mainsite/settings.py [hunk 1]
```diff
@@ -208,8 +212,10 @@
 ##
 
 # CORS_ORIGIN_ALLOW_ALL = True
-CORS_URLS_REGEX = r'^.*$'
+COR_URLS_REGEX = r'^.*$'
```
[the remark this answers is not in the document]
Comment (timber-they) [reply apps/mainsite/settings.py]: Huh, I do think so

File: apps/oidc/oeb_oidc_authentication_backend.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        user = super(OebOIDCAuthenticationBackend, self).create_user(claims)
+
+        user.first_name = 'unknown'
+        user.last_name = 'unknown'
+        user.email = convertSubToMail(claims.get('sub'))
+        user.set_email_items([{
+            'primary': True,
+            # Set this to verified since this is required for the user to be able to login.
+            # Note that this kinda breaches our idea of always having a verified mail;
+            # Since the mail actually only is a dummy mail. TODO: Handle this case
```
[the remark this answers is not in the document]
Comment (timber-they) [reply apps/oidc/oeb_oidc_authentication_backend.py:44]: I think we already have it on our radar

File: apps/oidc/oidc_views.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+class OidcView():
+    @api_view(['GET'])
+    @authentication_classes([])
+    @permission_classes([])
+    def oidcLogoutRedirect(req):
+        if req.method != 'GET':
+            return JsonResponse({"error": "Method not allowed"}, status=status.HTTP_405_METHOD_NOT_ALLOWED)
+
+        # TODO: Currently the automatic logout / redirect doesn't work, since we
```
[the remark this answers is not in the document]
Comment (timber-they) [reply apps/oidc/oidc_views.py:17]: Sven knows about this and it's not decided that we do in fact want to change something. I think the TODO here still makes sense to focus the attention of people later implementing it there.

**Pull request merged**